src/main/java/org/opentripplanner/routing/edgetype/VehicleRentalEdge.java

```diff
@@ -135,17 +134,18 @@ public boolean hasBogusName() {
     }
 
     /**
-     * @param stationNetworks The station where we want to drop the bike off.
-     * @param rentedNetworks The set of networks of the station we rented the bike from.
+     * @param stationNetwork The station network where we want to drop the bike off.
+     * @param rentedNetwork The networks of the station we rented the bike from.
      * @return true if the bike can be dropped off here, false if not.
      */
-    private boolean hasCompatibleNetworks(Set<String> stationNetworks, Set<String> rentedNetworks) {
+    private boolean hasCompatibleNetworks(String stationNetwork, String rentedNetwork) {
         /*
-         * Two stations are compatible if they share at least one network. Special case for "null"
-         * networks ("catch-all" network defined).
+         * Special case for "null" networks ("catch-all" network defined).
          */
-        if (stationNetworks == null || rentedNetworks == null)
```

Comment:
Is it possible that there are stations with null network and stations with network information at the same time or should all stations have network information if one has?

Reply:
`stationNetworks` is never null. It could be previously, if an OSM station did not have it set. `rentedNetworks` however can be null, when travelling backwards in time, searching for a floating vehicle.